crates/uv-cli/src/lib.rs

```diff
@@ -2731,7 +2731,7 @@ pub struct RunArgs {
     /// Omit other dependencies. The project itself will also be omitted.
     ///
     /// This option is an alias for `--only-group dev`.
-    #[arg(long, conflicts_with("no_dev"))]
+    #[arg(long, conflicts_with_all = ["group", "dev", "all_groups"])]
     pub only_dev: bool,
```

Comment:
`--only-dev` and `--dev` do not conflict, `--no-dev`?
```suggestion
    #[arg(long, conflicts_with_all = ["group", "no_dev", "all_groups"])]
    pub only_dev: bool,
```

Reply:
The idea I'm trying to express here is essentially that `--dev` and `--only-dev` is "what are you even doing". 

This is more clear with `--group A` and `--only-group B`, which is clearly "what no", but rejecting that and *not* rejecting `--group A --only-group A` is a pain for very little benefit. Since `--dev --only-dev` essentially desugars to that, I figured it's best to reject it for uniformity (and the followup code essentially assumes only one of these flags is ever true).

Reply:
That makes sense. I'd prefer to just allow redundant options, but understand it can get complicated and if it's not worth it it's not a big deal. Is it a breaking change though?

`--only-dev --no-dev` should still error though, no? Or does it not because we say that "exclusions override" and that's fine?

Reply:
Yeah I wasn't sure how exclusions and overrides interact. It should probably be tightened up but I didn't want to break something (I'll test it out / read the docs properly to figure out how this should be handled).

In terms of breaking... hmm... seems like the old impl tried harder to rationalize both inputs being provided... but in a bit of an adhoc way... 

https://github.com/astral-sh/uv/blob/ac1004284acfe3e3b0a7cce08741c8396dafc964/crates/uv-configuration/src/dev.rs#L234-L256

Reply:
In general, I'd say a breaking change here would not be ideal. I'd suggest PR'ing some tests for edge-cases to `main` first so we can see if they change here? If it seems hard to avoid a regression, then I guess we should wait for 0.6?

Reply:
Ok cool so this is the Final Potential Breakage to hash out (collapsed a few others that I regard as Equivalent).

Looking closer, the current code:
* Hard panics(!) on `--group --only-dev` and `--dev --only-group`
* CLI conflicts  `--group --only-group`
* CLI conflicts `--dev --only-dev`
* Saturates `--only-dev --dev` to `--only-dev`
* CLI conflicts `--only-dev --no-dev`
* Positionally Overrides `--dev --no-dev --dev ...`

So overall the current code hard rejects trying to mix `--only-*` with `--group`/`--dev`, except in the exact case of `--only-dev --dev`, which it turns into `--only-dev`. Which is fair enough, if a bit inconsistent. It's interesting that there's no overriding between `--only-dev and --no-dev` but I don't want to poke that bear in this PR.

I've adjusted things to preserve this behaviour, although the hard panics are now CLI conflicts.